### PR TITLE
[main] Update ClusterConfig with management.*ClusterConfigSpec

### PR DESCRIPTION
**Pull request by valaparthvi**
Fixes: #274 

Changes introduced with the PR:
1. Update GKE ClusterConfig and HostedClusterConfig with new fields from management.GKEClusterConfigSpec
2. Update AKS ClusterConfig and HostedClusterConfig with new fields from management.AKSClusterConfigSpec
3. Update EKS ClusterConfig and HostedClusterConfig with new fields from management.EKSClusterConfigSpec
4. Add `.idea` to .gitignore.

### CODE REVIEW

File: extensions/clusters/aks/aks_cluster_config.go [hunk 1]
```diff
@@ -11,15 +11,19 @@ const (
...
+	BaseURL                     *string           `json:"baseUrl,omitempty" yaml:"baseUrl,omitempty"`
 	DNSPrefix                   *string           `json:"dnsPrefix,omitempty" yaml:"dnsPrefix,omitempty"`
+	HTTPApplicationRouting      *bool             `json:"httpApplicationRouting,omitempty" yaml:"httpApplicationRouting,omitempty"`
 	KubernetesVersion           *string           `json:"kubernetesVersion,omitempty" yaml:"kubernetesVersion,omitempty"`
 	LinuxAdminUsername          *string           `json:"linuxAdminUsername,omitempty" yaml:"linuxAdminUsername,omitempty"`
 	LinuxSSHPublicKey           *string           `json:"sshPublicKey,omitempty" yaml:"sshPublicKey,omitempty"`
 	LoadBalancerSKU             *string           `json:"loadBalancerSku,omitempty" yaml:"loadBalancerSku,omitempty"`
 	LogAnalyticsWorkspaceGroup  *string           `json:"logAnalyticsWorkspaceGroup,omitempty" yaml:"logAnalyticsWorkspaceGroup,omitempty"`
 	LogAnalyticsWorkspaceName   *string           `json:"logAnalyticsWorkspaceName,omitempty" yaml:"logAnalyticsWorkspaceName,omitempty"`
+	ManagedIdentity             **bool            `json:"managedIdentity,omitempty" yaml:"managedIdentity,omitempty"`
```
Comment (caliskanugur) [inline extensions/clusters/aks/aks_cluster_config.go:26]: Is this the expected type for this field?

Comment (igomez06) [reply extensions/clusters/aks/aks_cluster_config.go:26]: my only question as well.

Comment (valaparthvi) [reply extensions/clusters/aks/aks_cluster_config.go:26]: I had the same question :sweat: But that's what I noticed [here](https://github.com/rancher/rancher/blob/7c36ac3b730c4733434269bda75e2906955ef22c/pkg/client/generated/management/v3/zz_generated_aks_cluster_config_spec.go#L56), so I copied it, just like all the other fields that I added.

Comment (valaparthvi) [reply extensions/clusters/aks/aks_cluster_config.go:26]: @thardeck perhaps you can help with this. I see you are the author of https://github.com/rancher/rancher/pull/42388. Is `**bool` valid or is something amiss here? I noticed the code in the PR is autogenerated, but I am unsure about the source.

Comment (igomez06) [reply extensions/clusters/aks/aks_cluster_config.go:26]: It was more of a question, if this done in the generated code then this should be good to go. No need to involve Tim.

Comment (valaparthvi) [reply extensions/clusters/aks/aks_cluster_config.go:26]: > It was more of a question, if this done in the generated code then this should be good to go. No need to involve Tim.

@igomez06 @caliskanugur In that case can we merge the PR? It is kinda blocking my work.

**Review by caliskanugur — APPROVED** [review]
Looks good to me

**Comment by valaparthvi** [comment]
> Do we need backports for this? or does only the main branch need the changes?

A backport to release/v2.9, I think.

**Comment by valaparthvi** [comment]
@igomez06 Can you give this a final review? TIA :)

**Review by igomez06 — APPROVED** [review]
LGTM

**Comment by valaparthvi** [comment]
>@valaparthvi yes, are the backports ready?

Since this PR has multiple commits, I decided to cherry-pick the squashed commit and create the backport. Would that be okay?

**Comment by igomez06** [comment]
> > @valaparthvi yes, are the backports ready?
> 
> Since this PR has multiple commits, I decided to cherry-pick the squashed commit and create the backport. Would that be okay?
please squash the commit here too, and cherry pick is great. Thank you!

**Comment by valaparthvi** [comment]
Backport release/v2.9 - https://github.com/rancher/shepherd/pull/287

**Pull request merged**